typing/typecore.ml

```diff
@@ -845,6 +862,40 @@ let wrap_disambiguate msg ty f x =
   | Wrong_name_disambiguation (env, wrong_name) ->
     raise (Error (wrong_name.name.loc, env, Wrong_name (msg, ty, wrong_name)))
 
+(* Use a dummy instance of [NameChoice] to disambiguate between [float array]
+   and [floatarray]. Note that an actual ambiguous choice is not possible, as we
+   default to ['a array] in case of doubt. *)
+
+module FloatArray = NameChoice(struct
+    type t = bool
+    type usage = unit
+    let kind = Datatype_kind.Record
+    let get_name = function
```

Comment:
I think this function should just be the constant function returning something like `"([||])"`. It should match the longident that is passed into disambiguate. If you expand the patch to disambiguate array projection as well -- which I think you should -- then using `"(.())"` would make the most sense.